src/games/strategy/triplea/ui/TripleAFrame.java

```diff
@@ -283,7 +284,7 @@ private void showCommentLog() {
         uiContext.getMapData().getMapDimensions().height);
     final Image small = uiContext.getMapImage().getSmallMapImage();
     smallView = new MapPanelSmallView(small, model);
-    mapPanel = new MapPanel(data, smallView, uiContext, model);
+    mapPanel = new MapPanel(data, smallView, uiContext, model, this);
```

Comment:
Introducing a circular dependency here.
Can be avoided by replacing the `this` with a function:

```
  public MapPanel(final GameData data, final MapPanelSmallView smallView, final IUIContext uiContext,
 -      final ImageScrollModel model) {
 +      final ImageScrollModel model, Function<MouseEvent, Integer> scrollSpeedCalculator) 
```

Then when calling the constructor:
`e ->  tripleAFrame.computeScrollSpeed(e)`